### PR TITLE
Fix: Make `domain` and `whiteList`, `protocalWhiteList` in the same lower case

### PR DESCRIPTION
**Pull request by ghost**
Change Logs:

1) securities.js：Add convert `whiteList` and `protocalWhiteList` to the
`lowerCase()`
in union.
2) context.js: Remove undefined condition, because this has been done in
`securities.js`.
3) surl.js: Make the header in the same `lowerCase()`.
4) hsts.js: Remove the comment's 1st line's empty space at the end (auto fixed by eslint).
5) utils.test.js: Change comments by fixing typos.
6) test\fixtures\apps\isSafeDomain\app\router.js: Add more unit tests to
test with.
7) test\fixtures\apps\isSafeDomain\config\config.js: Add more
domainWhiteList items for testing.

### CODE REVIEW

**Comment by ghost** [comment]
/cc：@dead-horse。This will fix the rest problem when your domain doesn't match domain_white_list. So as for `protocalWhiteList` (make them in `lowerCase()` in union).

Thanks for your time of reviewing!

File: app/middleware/securities.js [hunk 1]
```diff
@@ -10,6 +10,25 @@ module.exports = (_, app) => {
   const middlewares = [];
   const defaultMiddleware = (options.defaultMiddleware || '').split(',');
 
+  // Make sure that `whiteList` or `protocolWhiteList` is case insensitive
+  options.domainWhiteList = options.domainWhiteList || [];
```
Comment (dead-horse) [inline app/middleware/securities.js:14]: options.domainWhiteList = options.domainWhiteList.map(domain => domain.toLowerCase());

Comment (ghost) [reply app/middleware/securities.js:14]: OK, Fixed

**Comment by Maledong** [comment]
@dead-horse：Not sure why `continuous-integration/travis-ci/pr` is out of time. So I submitted again and have a try :)

**Comment by ghost** [comment]
@dead-horse : All is fixed.

**Review by dead-horse — APPROVED** [review]

**Comment by dead-horse** [comment]
LGTM

**Comment by ghost** [comment]
Thanks anyway!

**Pull request merged**